### PR TITLE
combine components in main.py

### PR DESCRIPTION
**Pull request by Theskrtnerd**

### CODE REVIEW

**Comment by thaohp113** [comment]
## Check requirement

- **Design/plan**: currently there’s no clear overview of design/plan documentation → ***Next step***: must add design/plan (ex: which techstacks have been used, tasks,… - adding either design or plan is fine I guess)
- **Testing:** Unit test is not compulsory, but since your team has written some, here are some comments:
    - Run poetry run pytest return an error at file tests/test_file_handler.py at test case 2 because it’s not compatible with linux → ***Next step:*** should handle this test case for Linus, although it has passed the CI test
    - Test image is not suitable since there’s no text on the image

Screenshot 2024-07-04 at 10 23 32 AM 

- **Output:** The last 2 pull requests have not been merged, and no reviewer has been added yet. → ***Next step:*** review the team’s work flow and merge these two PRs into main branch
    
    
Screenshot 2024-07-04 at 10 25 17 AM

## Code convention

- **Documentation**: there’s no clear and consistent documentation for code - no docstrings, some functions have comments, but some other don’t (the reference is from `file_handler.py` → ***Next step***: fix documentation and add comment
Screenshot 2024-07-04 at 10 25 48 AM

- **Some part is not clean yet**: some codes are currently commented out at `ocr_parser.py` - which is quite confusing (guess this is because your team has yet finalized the code) → ***Next step***: finalize and clean these codes.
    
Screenshot 2024-07-04 at 10 27 12 AM

**Review by hungpham3112 — CHANGES_REQUESTED** [review]
You should add two more flags in CLI  for specific input and output instead of forcing users to use /input and /output.

**Comment by Theskrtnerd** [comment]
> ## Check requirement
> * **Design/plan**: currently there’s no clear overview of design/plan documentation → _**Next step**_: must add design/plan (ex: which techstacks have been used, tasks,… - adding either design or plan is fine I guess)
> * **Testing:** Unit test is not compulsory, but since your team has written some, here are some comments:
>   
>   * Run poetry run pytest return an error at file tests/test_file_handler.py at test case 2 because it’s not compatible with linux → _**Next step:**_ should handle this test case for Linus, although it has passed the CI test
>   * Test image is not suitable since there’s no text on the image
>   * Some test files are blank
> 
> Screenshot 2024-07-04 at 10 23 32 AM
> * **Output:** The last 2 pull requests have not been merged, and no reviewer has been added yet. → _**Next step:**_ review the team’s work flow and merge these two PRs into main branch
> 
> Screenshot 2024-07-04 at 10 25 17 AM
> ## Code convention
> * **Documentation**: there’s no clear and consistent documentation for code - no docstrings, some functions have comments, but some other don’t (the reference is from `file_handler.py` → _**Next step**_: fix documentation and add comment
> 
> Screenshot 2024-07-04 at 10 25 48 AM
> * **Some part is not clean yet**: some codes are currently commented out at `ocr_parser.py` - which is quite confusing (guess this is because your team has yet finalized the code) → _**Next step**_: finalize and clean these codes.
> 
> Screenshot 2024-07-04 at 10 27 12 AM

Thank you so much for your detailed comment. For your suggestions, I have done the following:

* Design/plan: I've added our design plan in the repo (`design-plan.png`)
* Testing:
   * For the Linux test case, we will try to think of a different solution as our current Python module does not support that OS.
   * I've changed the test image to another one with text in it.
   * We will add the other test files later.
* Output: The 2 open PRs are meant to be reviewed before merging, and I have requested your team for review. Please give at least 2 reviews in order to merge the PR.
* Code convention: We will add the comments and clean the codebase later
*

File: llamagon_ocr_script/main.py [hunk 1]
```diff
@@ -32,7 +30,13 @@ def main(file_name, upload):
     parser.add_argument(
         "--upload", action="store_true", help="Upload the result folder to the cloud."
     )
+    parser.add_argument(
+        "--input_folder", type=str, default="input", help="The input folder containing the PDF file."
+    )
+    parser.add_argument(
+        "--output_folder", type=str, default="output", help="The folder to save the output images and text."
+    )
```
Comment (hungpham3112) [inline llamagon_ocr_script/main.py:45]: where is the file_name parser? You used `args.file_name` but didn't have an argument for it.

**Review by hungpham3112 — CHANGES_REQUESTED** [review]

**Review by thaohp113 — APPROVED** [review]

**Review by hungpham3112 — APPROVED** [review]

**Pull request merged**